api/envoy/extensions/filters/common/set_filter_state/v3/rule.proto

```diff
@@ -0,0 +1,52 @@
+syntax = "proto3";
+
+package envoy.extensions.filters.common.set_filter_state.v3;
+
+import "envoy/config/core/v3/substitution_format_string.proto";
+
+import "udpa/annotations/status.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.filters.common.set_filter_state.v3";
+option java_outer_classname = "RuleProto";
+option java_multiple_files = true;
+option go_package = "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/common/set_filter_state/v3;set_filter_statev3";
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: Set-Filter-State Update Rule]
+
+// A filter state update rule.
+// [#next-free-field: 6]
+message Rule {
+  enum SharedWithUpstream {
+    // Object is not shared with the upstream internal connections.
+    NONE = 0;
+
+    // Object is shared with the upstream internal connection.
+    ONCE = 1;
+
+    // Object is shared with the upstream internal connection and any internal connection upstream from it.
+    TRANSITIVE = 2;
+  }
+
+  // Filter state key.
+  string key = 1 [(validate.rules).string = {min_len: 1}];
```

Comment:
Drive-by comment due to the conversation over the issue.
Initially what I thought is that this filter lets you set FS string objects, after recapping with latest object factory additions I understand what your intention was here.
However, I believe there's a place to support more options here, i.e. setting an object from factory, or a plain string (using ``Router::StringAccessorImpl``) or integers, (using ``StreamInfo::Uint32AccessorImpl``).

What do you say about wrapping ``key`` in ``oneof`` and renaming ``key`` to-> ``object_key``, so this can later be extended to ``string_key`` or ``integer_key``, to support the use-case I mentioned above?

My motivation is having a way to set FS values without any necessity to write custom code or object factory. For example - reading a file and setting the file content in a filter state object, this should be possible since the ``SubstitutionFormatString`` supports file source.

Thanks

Reply:
Yeah, that makes sense. Wasm also uses a specific CelState type that is not unique by key, so I am aware of the problem. Will update.

Reply:
Done.

Reply:
Thanks